### DISCUSSION TITLE
Invalid Marker value

### DISCUSSION THREAD

**Discussion by am-tc01** [idaholab/moose#27613] [Q&A Meshing]
Hi,
I have implemented a modified ErrorToleranceMarker. It pretty much works in the same way as the original ErrorToleranceMarker plus I have few other `if` conditions which check the current level (`_current_elem->level()`) of the element and Refine/Coarsen the element based on that. Apart from that, I also run an additional adaptivity step with an increased value of `max_h_level`, in case a solution step fails. In most of the cases, it works fine but sometimes I see the following error (the value is of course different every time).
`Invalid Marker value detected: 0.8125`
I have tried to debug it but couldn't find the exact reason. Any clue what is causing this?

Regards.

**Comment by GiudGiud** [comment]
Hello

I think Marker values are supposed to be 0 or 1 or -1. Nothing in between
This comes from there
```
    // round() is a C99 function, it is not located in the std:: namespace.
    marker_value = static_cast<Marker::MarkerValue>(round(dof_value));

    // Make sure we aren't masking an issue in the Marker system by rounding its values.
    if (std::abs(marker_value - dof_value) > TOLERANCE * TOLERANCE)
      mooseError("Invalid Marker value detected: ", dof_value);
```

so maybe you need to change your marker to force it to abide by these rounded values

Guillaume

**Reply by am-tc01** [reply]
Thanks for your answer. I return one of the following from the marker, `REFINE`, `COARSEN` or `DO_NOTHING` and I guess they correspond to 1, -1 and 0. I don't understand why it returns a value other than that. It happens when I run an additional adaptivity step on failure.

**Reply by GiudGiud** [reply]
I don't know either
Do you have something else writing to this variable? Maybe an auxkernel?
